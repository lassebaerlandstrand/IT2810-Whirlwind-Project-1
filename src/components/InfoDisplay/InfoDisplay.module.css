.weatherDetailsContainer {
  display: flex;
  justify-content: space-around;
  margin: 3rem auto;
  flex-wrap: wrap;
  padding: 1rem;
  gap: 1rem;
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  list-style: none;
}

.listItemContainer {
  flex: 1 1 12rem;
  min-width: 12rem;
  max-width: 20rem;
}

.container {
  display: flex;
  justify-content: center;
  margin: 0 auto;
  background-color: var(--secondary-color);
  border-radius: 1rem;
  flex-wrap: wrap;
  padding: 0.5rem 0.5rem 3rem 0.5rem;
}

.iconContainer {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2rem;
  height: 2rem;
  border-radius: 100%;
  background-color: var(--tertiary-color);
  transition: 'transform 2s';
  svg {
    stroke: #ffffff;
    stroke-width: 1;
    fill: #c1d6db;
  }
}

.title {
  font-size: 1.3rem;
  font-weight: normal;
  min-width: 6rem;
}

.titleContainer {
  width: 80%;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  gap: 0.5rem;
}

.infoContainer {
  width: 10rem;
  height: 10rem;

  background-color: var(--primary-color);
  border-radius: 100%;

  flex-wrap: wrap;

  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--secondary-color);
  gap: 0.5rem;
}

.infoText {
  font-size: 2rem;
}

.infoUnit {
  font-size: 1rem;
}

/* Cannot use CSS variables in media query. Therefore have to specify --breakpoint-md explicitly */
@media screen and (min-width: 768px) {
  .weatherDetailsContainer {
    width: 100%;
    margin: 3rem 0 0 0;
    justify-content: space-between;
  }
}
