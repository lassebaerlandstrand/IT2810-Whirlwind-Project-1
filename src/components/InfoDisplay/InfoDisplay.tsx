import styles from './InfoDisplay.module.css';

interface InfoDisplayItemProps {
  icon: React.ReactNode;
  infoText: string;
  amount: number | undefined;
  unit: string;
  rotation: number | undefined;
}

export const InfoDisplayItem = ({ icon, infoText, amount, unit, rotation }: InfoDisplayItemProps) => {
  return (
    <div className={styles.container}>
      <div className={styles.titleContainer}>
        <div
          className={styles.iconContainer}
          style={{
            transform: `rotate(${rotation !== undefined ? rotation : 0}deg)`,
            transition: 'transform 2s',
          }}
        >
          {icon}
        </div>
        <h3 className={styles.title}>{infoText}</h3>
      </div>
      <div className={styles.infoContainer}>
        <p className={styles.infoText}>{amount !== undefined ? amount.toString() : '--.-'}</p>
        <p className={styles.infoUnit}>{unit}</p>
      </div>
    </div>
  );
};

interface InfoDisplayProps {
  infos: InfoDisplayItemProps[];
}

const InfoDisplay = ({ infos }: InfoDisplayProps) => {
  return (
    <ul className={styles.weatherDetailsContainer}>
      {infos.map((item, index) => (
        <li key={index} className={styles.listItemContainer}>
          <InfoDisplayItem
            icon={item.icon}
            infoText={item.infoText}
            amount={item.amount}
            unit={item.unit}
            rotation={item.rotation}
            key={index}
          />
        </li>
      ))}
    </ul>
  );
};

export default InfoDisplay;
