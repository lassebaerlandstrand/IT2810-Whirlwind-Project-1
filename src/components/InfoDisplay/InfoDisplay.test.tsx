import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import InfoDisplay from './InfoDisplay';

describe('Infodisplay tests', () => {
  it('checks that the component renders properly with basic inputs', () => {
    render(<InfoDisplay icon={undefined} infoText={'Rain'} amount={10} unit={'mm'} rotation={0} />);

    expect(screen.getByText('Rain')).toBeInTheDocument();
    expect(screen.getByText('10')).toBeInTheDocument();
    expect(screen.getByText('mm')).toBeInTheDocument();
  });

  it('checks that the component loads properly when loading', () => {
    render(<InfoDisplay icon={undefined} infoText={''} amount={undefined} unit={''} rotation={undefined} />);

    // Check if amount displays --.- when it lacks info
    expect(screen.getByText('--.-')).toBeInTheDocument();
  });
});
