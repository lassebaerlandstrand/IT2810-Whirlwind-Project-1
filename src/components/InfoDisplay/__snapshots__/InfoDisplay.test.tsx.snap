// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoDisplay tests > should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="_weatherDetailsContainer_7869c2"
  >
    <li
      class="_listItemContainer_7869c2"
    >
      <div
        class="_container_7869c2"
      >
        <div
          class="_titleContainer_7869c2"
        >
          <div
            class="_iconContainer_7869c2"
            style="transform: rotate(34deg); transition: transform 2s;"
          >
            <svg
              class="tabler-icon tabler-icon-wind "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 8h8.5a2.5 2.5 0 1 0 -2.34 -3.24"
              />
              <path
                d="M3 12h15.5a2.5 2.5 0 1 1 -2.34 3.24"
              />
              <path
                d="M4 16h5.5a2.5 2.5 0 1 1 -2.34 3.24"
              />
            </svg>
          </div>
          <h3
            class="_title_7869c2"
          >
            Wind (NA)
          </h3>
        </div>
        <div
          class="_infoContainer_7869c2"
        >
          <p
            class="_infoText_7869c2"
          >
            5
          </p>
          <p
            class="_infoUnit_7869c2"
          >
            m/s
          </p>
        </div>
      </div>
    </li>
    <li
      class="_listItemContainer_7869c2"
    >
      <div
        class="_container_7869c2"
      >
        <div
          class="_titleContainer_7869c2"
        >
          <div
            class="_iconContainer_7869c2"
            style="transform: rotate(0deg); transition: transform 2s;"
          >
            <svg
              class="tabler-icon tabler-icon-cloud-rain "
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 18a4.6 4.4 0 0 1 0 -9a5 4.5 0 0 1 11 2h1a3.5 3.5 0 0 1 0 7"
              />
              <path
                d="M11 13v2m0 3v2m4 -5v2m0 3v2"
              />
            </svg>
          </div>
          <h3
            class="_title_7869c2"
          >
            Precipitation
          </h3>
        </div>
        <div
          class="_infoContainer_7869c2"
        >
          <p
            class="_infoText_7869c2"
          >
            4
          </p>
          <p
            class="_infoUnit_7869c2"
          >
            mm/h
          </p>
        </div>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
